examples/demo/src/main/scala/Main.scala

```diff
@@ -43,6 +45,53 @@ object Main {
       DataRow().addCell("school", "Well Known School").addCell("numStudents", 30000)
     )
 
+    val recordHeaders: List[DataRecord] = List(
+      DataRecord()
+        .addCell("dateHeader", "Date")
+        .addRecord(List(
+          DataCell("peerCloseHeader", "AAPL close"),
+          DataCell("peerChangeHeader", "AAPL")
+          ))
+        .addRecord(List(
+          DataCell("peerCloseHeader", "MSFT close"),
+          DataCell("peerChangeHeader", "MSFT")
+          ))
+        .addRecord(List(
+          DataCell("peerCloseHeader", "GOOG close"),
+          DataCell("peerChangeHeader", "GOOG")
+          ))
+
+    val recordData: List[DataRecord] = List(
+      DataRecord()
+        .addCell("date", LocalDate.of(2020, MAY, 1))
+        .addRecord(List(
+          DataCell("peerClose", 1.08),
+          DataCell("peerChange", 0.01)
+          ))
+        .addRecord(List(
+          DataCell("peerClose", 2.31),
+          DataCell("peerChange", -0.01)
+          ))
+        .addRecord(List(
+          DataCell("peerClose", 4.25),
+          DataCell("peerChange", 0.02)
```

Comment:
What happens if they add an additional `peerClose, peerChange` record so that there are 4 pairs of data provided but we only expect 3 pairs of data from the headers? Do we keep expanding the columns?

Reply:
Yeah, good point.

Right now I tend to think that the best and most incremental approach is to keep things simple on the Exscalabur side: add a validation so that Exscalabur is strict about the number of headers matching the number of data cells.

In case that's violated, send an error back to the client.

This allows us to deliver functionality now, while we iterate and learn what's the best way to report specification in Exscalabur.

Reply:
actually, thinking about it now, I'm not sure this is an exscalabur related problem cause (at least right now), we don't make any associations between any two rows of the output workbook. In that case, providing more columns than we have headers for is, at least from Exscalabur's POV, equivalent to having two data columns of unequal length. In which case, if they want the two to always be the same length, I think that should be an assertion on the client's end, not Exscalabur's.

At least right now, data-length validation doesn't seem to be something exscalabur should be concerned about.

Reply:
Yeah I think you should be allowed to write whatever you want into Exscalabur. The number of columns in each row should be dependent on the data in the row not some other row (even if that other row is a table header).